zos-utils/zossrc/javaUsermap.c

```
@@ -0,0 +1,122 @@
+    }
+    char *output = (char*) __malloc31(length + 1);
+    strncpy(output, ebcdic, length);
+    output[length] = 0;
+
+    int size = __etoa(output);
+    if (size < 0) {
+        jclass exception_clazz = (*env) -> FindClass(env, JNI_CLASS_ILLEGAL_ARGUMENT_EXCEPTION);
+        (*env) -> ThrowNew(env, exception_clazz, JNI_MESSAGE_CANNOT_CONVERT_USER_ID);
+        free(output);
+        return NULL;
+    }
+
+    jstring outputJstring = (*env) -> NewStringUTF(env, output);
+    free(output);
+    return outputJstring;
+}
+
+JNIEXPORT jobject JNICALL Java_org_zowe_commons_usermap_UserMapper_getUserIDForCertificate(JNIEnv *env, jobject obj, jbyteArray certificate) {
+
+    jbyte* cCertificate = (*env)->GetByteArrayElements(env, certificate, NULL);
+    int certificateLength = (*env)->GetArrayLength(env,certificate);
+    char useridRacf[9] = {0};
+    int userId_length = 9;
+    int returnCodeRacf = 0;
+    int reasonCodeRacf = 0;
+
+    int rc = __certificate(__CERTIFICATE_AUTHENTICATE, certificateLength, (char*)cCertificate,userId_length,useridRacf);
+    (*env)->ReleaseByteArrayElements(env, certificate, cCertificate, 0);
+
+    if ((*env) -> ExceptionCheck(env)) {
+      return NULL;
+    }
+    e2a(useridRacf,9);
+
+    jclass certificateClass = (*env)->FindClass(env,JNI_CLASS_CERTIFICATE_RESPONSE);
+
+    jmethodID cid = (*env)->GetMethodID(env,certificateClass,JNI_METHOD_CONSTRUCTOR,JNI_SIGNATURE_METHOD_STRING_INT_INT_INT_VOID);
+
+    jstring jUseridRacf = (*env)->NewStringUTF(env,useridRacf);
+
+    return (*env)->NewObject(env,certificateClass,cid,jUseridRacf,rc,errno,__errno2());
+}
+
+JNIEXPORT jobject JNICALL Java_org_zowe_commons_usermap_UserMapper_getUserIDForDN(JNIEnv *env, jobject obj, jstring dn, jstring reg){
+    const char* distName = (*env)->GetStringUTFChars(env,dn,NULL);
+    int dnLength = (*env)->GetStringUTFLength(env,dn);
+    char distinguishedName[246] = {0};
+    memcpy(distinguishedName,distName,dnLength);
```

there should be a check to compare dnLength and 246. In the case of exceeding the resources rather throw an exception.

---

add if